### PR TITLE
Zadanie 1 - 3.2, 3.3

### PR DESCRIPTION
**Pull request by maybegamer**
215693

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java [hunk 1]
```diff
@@ -4,21 +4,25 @@
...
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+import edu.kis.powp.jobs2d.features.DriverFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DrawPanelAdapter implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java]: The name lacks 'driver' aspect. (Adapter is only an implementation detail)

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java [hunk 2]
```diff
@@ -4,21 +4,25 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+import edu.kis.powp.jobs2d.features.DriverFeature;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java]: Unused imports?

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java [hunk 3]
```diff
@@ -27,11 +31,13 @@ public void operateTo(int x, int y) {
...
-		drawLine(line);
+		setPosition(x, y);
+
+		drawPanelController.drawLine(line);
 	}
 
 	@Override
 	public String toString() {
-		return "@Q!$!@$!#@$(*#@&Q(%^*#@";
+		return "Buggy Simulator";
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java]: it is not "buggy" now

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java [hunk 4]
```diff
@@ -16,6 +16,11 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+
+		if (e.getActionCommand().equals("Figure Joe 1")) {
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java]: Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.

**Review by kowallus — CHANGES_REQUESTED** [review]
Specific remarks are inlined in the code.
They can be fixed during labs (today).
UML diagram has incorrect aggregation (direction). 
About answer: the question concerned exclusively the CLASS variant of the adapter pattern (not the intern implementation).

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java (hunk 3 above)
Comment (maybegamer) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java]: Fixed

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java (hunk 4 above)
Comment (maybegamer) [reply src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java]: Fixed

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java (hunk 2 above)
Comment (maybegamer) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java]: Fixed

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java (hunk 1 above)
Comment (maybegamer) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawPanelAdapter.java]: Fixed

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2dDriverAdapter.java [hunk 5]
```diff
@@ -4,17 +4,15 @@
...
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
-import edu.kis.powp.jobs2d.features.DrawerFeature;
-import edu.kis.powp.jobs2d.features.DriverFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class DrawPanelAdapter implements Job2dDriver {
+public class Job2dDriverAdapter implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2dDriverAdapter.java:11]: Now the name does not indicate the purpose of the adapter or what was actually adapted.

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java [hunk 6]
```diff
@@ -3,21 +3,24 @@
...
+	public SelectTestFigureOptionListener(DriverManager driverManager, DrawingType drawingType) {
 		this.driverManager = driverManager;
+		this.drawingType = drawingType;
 	}
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
 
-		if (e.getActionCommand().equals("Figure Joe 1")) {
+		if (drawingType == DrawingType.FIGURE_1) {
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:23]: FIGURE_2 not handled properly. Use switch.

**Review by kowallus — COMMENTED** [review]
Minor remarks.
Please also continue with starred (*) tasks.

**Pull request closed without merging**